test/adapter/json_api/resource_identifier_test.rb

```diff
@@ -22,62 +22,60 @@ class FragmentedSerializer < ActiveModel::Serializer; end
         end
 
         def test_defined_type
-          test_type(WithDefinedTypeSerializer, 'with_defined_type')
+          test(WithDefinedTypeSerializer.new(@model), type: 'with_defined_type')
         end
 
         def test_singular_type
-          test_type_inflection(AuthorSerializer, 'author', :singular)
+          test(AuthorSerializer.new(@model), type: 'author', inflection: :singular)
         end
 
         def test_plural_type
-          test_type_inflection(AuthorSerializer, 'authors', :plural)
+          test(AuthorSerializer.new(@model), type: 'authors', inflection: :plural)
+        end
+
+        def test_type_with_namespace
+          spam = Spam::UnrelatedLink.new
+          test(Spam::UnrelatedLinkSerializer.new(spam), type: 'spam--unrelated_links')
+        end
+
+        def test_type_with_custom_namespace
+          spam = Spam::UnrelatedLink.new
+          test(Spam::UnrelatedLinkSerializer.new(spam), type: 'spam/unrelated_links', namespace_separator: '/')
         end
 
         def test_id_defined_on_object
-          test_id(AuthorSerializer, @model.id.to_s)
+          test(AuthorSerializer.new(@model), id: @model.id.to_s)
         end
 
         def test_id_defined_on_serializer
-          test_id(WithDefinedIdSerializer, 'special_id')
+          test(WithDefinedIdSerializer.new(@model), id: 'special_id')
         end
 
         def test_id_defined_on_fragmented
           FragmentedSerializer.fragmented(WithDefinedIdSerializer.new(@author))
-          test_id(FragmentedSerializer, 'special_id')
+          test(FragmentedSerializer.new(@model), id: 'special_id')
         end
 
         private
 
-        def test_type_inflection(serializer_class, expected_type, inflection)
-          original_inflection = ActiveModelSerializers.config.jsonapi_resource_type
-          ActiveModelSerializers.config.jsonapi_resource_type = inflection
-          test_type(serializer_class, expected_type)
+        def test(serializer, opts = {})
```

Comment:
Maybe rename to `assert_something`?  given the way test::unit /minitest works, calling something `test`, even if private, triggers my spidey sense. Also, it would make it more clear what the assertion is.

(That said, I see the original code already has such private methods, so, :8ball: :eyes: :question: )